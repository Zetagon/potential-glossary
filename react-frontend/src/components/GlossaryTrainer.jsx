/**
 * @fileOverview Main file for glossary trainer
 * @name GlossaryTrainer.jsx
 * @author 
 * @license 
 */
/*jslint esversion:6*/
import React, { Component } from 'react';
import io from 'socket.io-client'
import logo from '../logo.svg';
import '../App.css';
import '../../public/css/client-glos.css';
import {DescriptionContainer} from './DescriptionContainer.jsx'
import {InputContainer, InputBox } from './InputContainer.jsx'
import CorrectionArea from './CorrectionArea.jsx'

class GlossaryTrainer extends Component {
  constructor(props){
    super(props);
    this.setCorrectionString = this.setCorrectionString.bind(this);
    this.state = {
      correctionMessage: "Correction!",
      descriptions: []
    }
    this.socket = io('/');
    this.socket.emit('getDescription');
    this.socket.on('getDescriptionResponse', (data) => {
      console.log('description received!' + data)
      this.setState({descriptions:data});
    })

  }

  setCorrectionString(correction){
    this.setState({correctionMessage:correction});
  }


  handleSubmit(){
    this.setCorrectionString('fel svar!');
  }

  render(){
    return (
      <div className="App">
        <DescriptionContainer
          descriptions={this.state.descriptions}
        />
        <InputContainer
          socket={this.socket}
          numberOfBoxes={ 5 }
          onClickProp={this.handleSubmit.bind(this)}
        />
        <CorrectionArea message={this.state.correctionMessage}/>
      </div>
    );
  }
}
export {GlossaryTrainer}
