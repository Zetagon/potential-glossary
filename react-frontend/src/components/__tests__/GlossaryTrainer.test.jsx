import React from 'react';
import renderer from 'react-test-renderer';
import { GlossaryTrainer } from '../GlossaryTrainer.jsx'


test('test',()=>{
    const component = renderer.create(
      <GlossaryTrainer/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.getInstance().setCorrectionString('this is a test string');
    tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
})
