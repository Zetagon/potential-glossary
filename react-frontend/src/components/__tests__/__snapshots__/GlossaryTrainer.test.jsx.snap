exports[`test test 1`] = `
<div
  className="App">
  <div
    className="description-container" />
  <div
    className="input-container">
    <div
      className="input-flex-container">
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
    </div>
    <div
      className="submit-button"
      onClick={[Function]}>
       Submit! 
    </div>
  </div>
  <div
    className="correction-container">
    <div>
      <div
        className="error-text">
         
        Correction!
         
      </div>
    </div>
  </div>
</div>
`;

exports[`test test 2`] = `
<div
  className="App">
  <div
    className="description-container" />
  <div
    className="input-container">
    <div
      className="input-flex-container">
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
      <div
        className="input-box"
        style={
          Object {
            "background": "white",
          }
        }>
        <input
          className="the-input"
          onKeyPress={[Function]}
          tabIndex="1" />
      </div>
    </div>
    <div
      className="submit-button"
      onClick={[Function]}>
       Submit! 
    </div>
  </div>
  <div
    className="correction-container">
    <div>
      <div
        className="error-text">
         
        this is a test string
         
      </div>
    </div>
  </div>
</div>
`;
